src/crypto/accounts.slice.ts

```diff
@@ -44,21 +42,31 @@ const slice = createSlice({
   name: sliceName,
   initialState,
   reducers: {
-    setAccountsToAdd(state, action: PayloadAction<SerializedWalletWithAddress[]>) {
-      state.accountsToAdd = action.payload;
+    setAddAccounts(
+      state,
+      action: PayloadAction<{
+        type: WalletType.PRIVATE_KEY | WalletType.MNEMONIC;
+        accounts: SerializedWalletWithAddress[];
+        secret: string;
+      }>
+    ) {
+      state.add = action.payload;
+    },
+    clearAddAccounts(state) {
+      state.add = undefined;
     }
   }
 });
 
-export const { setAccountsToAdd } = slice.actions;
+export const { setAddAccounts, clearAddAccounts } = slice.actions;
 
 export const addSavedAccounts = createAction<boolean>(`${sliceName}/addSavedAccounts`);
 
 export default slice;
 
 export const getAccountsToAdd = createSelector(
```

Comment:
Let's not export this to prevent confusion with the other function of the same name.